packages/api-folders/src/context/folders.ts

```diff
@@ -0,0 +1,167 @@
+/**
+ * Package mdbid does not have types.
+ */
+// @ts-ignore
+import mdbid from "mdbid";
+import WebinyError from "@webiny/error";
+import { NotFoundError } from "@webiny/handler-graphql";
+import joi from "joi";
+
+import {
+    Folder,
+    FolderInput,
+    FoldersConfig,
+    GetFolderParams,
+    IFolders,
+    ListFoldersParams
+} from "~/types";
+
+const requiredString = joi.string().required();
+
+const createSchema = joi.object({
+    name: requiredString.min(3),
+    slug: requiredString.min(3),
+    type: requiredString.pattern(/page|cms|file/),
+    tenant: requiredString,
+    locale: requiredString
+});
+
+const updateSchema = joi.object({
+    name: requiredString.min(3),
+    slug: requiredString.min(3)
+});
+
+export const createFoldersContext = async ({
+    getTenantId,
+    getLocaleCode,
+    getIdentity,
+    storageOperations
+}: FoldersConfig): Promise<IFolders> => {
+    const tenant = getTenantId();
+    const locale = getLocaleCode();
```

Comment:
Getting these values here will cause issues if you switch tenants during a single request lifecycle, because they will never get updated, and data will be stored to the wrong tenant. These functions should be called within each CRUD methods individually.